src/reactor.rs

```diff
@@ -552,6 +556,10 @@ mod sys {
             Ok(())
         }
         pub fn wait(&self, events: &mut Events, timeout: Option<Duration>) -> io::Result<usize> {
+            let timeout = timeout.map(|t| match t {
+                t if t == Duration::from_millis(0) || t >= Duration::from_millis(1) => t,
+                t if t < Duration::from_millis(1) => Duration::from_millis(1),
+            });
```

Comment:
This should be something like:

```rust
let timeout = timeout.map(|t| match t {
    if Duration::from_millis(0) < t && t < Duration::from_millis(1) {
        Duration::from_millis(1)
    } else {
        t
    }
});
```

Reply:
Or, even better:

```rust
let timeout = timeout.map(|t| match t {
    if t == Duration::from_millis(0) {
        t
    } else {
        cmp::max(t, Duration::from_millis(1))
    }
});
```

Reply:
And the epoll-specific logic could be:

```rust
let timeout_ms = timeout
    .and_then(|t| {
        if t == Duration::from_millis(0) {
            Some(t)
        } else {
            cmp::max(t, Duration::from_millis(1)).as_millis().try_into().ok()
        }
    });
```

Reply:
Or, if you like mutation :)

```rust
// epoll
let timeout_ms = timeout
    .and_then(|mut t| {
        if t > Duration::from_millis(0) {
            t = cmp::max(t, Duration::from_millis(1));
        }
        t.as_millis().try_into().ok()
    });

// wepoll
let timeout = timeout.map(|mut t| {
    if t > Duration::from_millis(0) {
        t = cmp::max(t, Duration::from_millis(1));
    }
    t
});
```

Reply:
If i may, could i ask you to set variables with durations in the very beginning of code and then use it where needed. Recently, i figured out that it may cost too much latency to calculate `Duration::from_xxx()` right in place.

Reply:
Updated

Reply:
@serzhiio It really doesn't matter, the calculation happens at compilation time and is zero instructions :)

https://play.rust-lang.org/?version=stable&mode=release&edition=2018

Reply:
Oh, in fact ... `Duration::from_xxx` functions are `const fn`s.

Reply:
so its definitely better to have a constant with the result :)
i'm experimenting with loops a lot and noticed some latency issues when using `Duration::from_xxx()` right inside, especially with very small durations micros and nanos. Havent dig it deeper.

Reply:
We're going into `epoll_wait` right after this, I can't see how it would make a difference.